src/services/db/GitFileSystemService.ts

```diff
@@ -479,6 +483,50 @@ export default class GitFileSystemService {
   // TODO: Delete a file
   async delete() {}
 
-  // TODO: Get the latest commit of branch
-  async getLatestCommitOfBranch() {}
+  isGitLocalDiskRawCommitData(
+    commit: unknown
+  ): commit is GitLocalDiskRawCommitData {
+    return (
+      !!commit &&
+      (commit as GitLocalDiskRawCommitData).author_name !== undefined &&
+      (commit as GitLocalDiskRawCommitData).author_email !== undefined &&
+      (commit as GitLocalDiskRawCommitData).date !== undefined &&
+      (commit as GitLocalDiskRawCommitData).message !== undefined &&
+      (commit as GitLocalDiskRawCommitData).hash !== undefined
+    )
+  }
+
+  getLatestCommitOfBranch(
+    repoName: string,
+    branchName: string
+  ): ResultAsync<GitLocalDiskCommitData, GitFileSystemError> {
+    return ResultAsync.fromPromise(
+      this.git.cwd(`${EFS_VOL_PATH}/${repoName}`).checkout(branchName).log(),
```

Comment:
Are we able to do a `.log()` and specify the branch directly here? Seems like it is possible to do a `git log staging` to get just the log of that branch without needing to checkout that branch.

Asking because checkouts tend to be quite expensive and it will involve modifying the state of the local repo, which may not be ideal.

Reply:
good point, will update!

Reply:
done